src/cclib/io/wfxwriter.py

```diff
@@ -0,0 +1,320 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2017, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""A writer for wfx format files."""
+
+import os.path
+import math
+from cclib.parser import utils
+
+from . import filewriter
+
+class WFXWriter(filewriter.Writer):
+    """A writer for wfx files."""
+
+    required_attrs = ['atomcoords', 'atomnos', 'gbasis', 'charge', 'homos', 'mult']
+    keyword = 'GTO'
+    num_orb = {'s':1, 'p':3, 'd':6, 'f':10, 'g':15, 'h':21,
+               'S':1, 'P':3, 'D':6, 'F':10, 'G':15, 'H':21}
+    count_orb = {'s':1, 'p':2, 'd':5, 'f':11, 'g':21, 'h':36,
+                 'S':1, 'P':2, 'D':5, 'F':11, 'G':21, 'H':36}
```

Comment:
It's not clear what these ```dicts``` are for only based on their names. Please rename to make more clear or add a comment. Also, seems like you could simplify and store just lowercase and access appropriately. A wrapper around ```dict``` that does the transformation would make sure it's always accessed correctly.